packages/flutter_tools/lib/src/run_cold.dart

```diff
@@ -131,6 +131,12 @@ class ColdRunner extends ResidentRunner {
       await connectToServiceProtocol();
     } catch (error) {
       printError('Error connecting to the service protocol: $error');
+      // https://github.com/flutter/flutter/issues/33050
+      // TODO(blasten): Remove this check once https://b.corp.google.com/issues/132325318 has been fixed.
+      if (error.toString().contains(kAndroidQHttpConnectionClosedExp)) {
```

Comment:
We might be able to reduce to scope of false positives here by looking at the device target platform

Reply:
I'm now checking for Android Q. :)